### PR TITLE
feat: guardian cli

### PR DESCRIPTION
**Pull request by rahmatrhd**

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -4,16 +4,31 @@ import (
...
 func Execute() {
+	var rootCmd = &cobra.Command{
+		Use: "guardian",
+	}
+
+	protoAdapter := v1.NewAdapter()
+
+	cliConfig, err := readConfig()
+	if err != nil {
+		fmt.Println(err)
```
Comment (StewartJingga) [inline cmd/root.go:21]: is this expected to proceed even when getting error of config reading?

File: cmd/config.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+var (
+	DefaultHost   = "localhost"
+	FileName      = ".guardian"
+	FileExtension = "yaml"
+)
+
+type config struct {
+	Host string `yaml:"host"`
```
Comment (StewartJingga) [inline cmd/config.go:18]: what's the difference between this config and app.Config?
does it make sense to merge them?

File: cmd/policies.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+		Use:   "update",
+		Short: "update policy",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			b, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+
+			var policy domain.Policy
+			switch filepath.Ext(filePath) {
```
Comment (StewartJingga) [inline cmd/policies.go:146]: we could extract this function so `createPolicyCommand` can use it too.

File: cmd/providers.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+	return cmd
+}
+
+func listProvidersCommand(c *config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "list",
+		Short: "list providers",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), time.Second*2)
```
Comment (StewartJingga) [inline cmd/providers.go:38]: I see these connection codes being used frequently, can we wrap it to a function?
e.g.
```
ctx, cancel := connectToHost(ctx)
```

File: cmd/config.go (hunk 2 above)
Comment (rahmatrhd) [reply cmd/config.go:18]: I was also having the same doubt, but the difference for cmd.config is that it is only the config for the CLI to run, while app.Config is more into the server's config

File: cmd/providers.go (hunk 4 above)
Comment (rahmatrhd) [reply cmd/providers.go:38]: it's a tricky one, we're deferring `dialCancel()` and `conn.Close()` from here only. I tried to put them into a function`createClient()` that also returns a function that calls both `dialCancel()` and `conn.Close()` for cancellation but it's not possible

**Review by StewartJingga — APPROVED** [review]
LGTM

**Pull request merged**